src/pal/src/thread/thread.cpp

```diff
@@ -2430,7 +2433,11 @@ PAL_GetStackBase()
     
     pthread_t thread = pthread_self();
     
+#if defined(__FreeBSD__)
```

Comment:
We should introduce a function probe into configure.cmake instead of guarding this on only FreeBSD.